src/zenml/cli/stack_components.py

```diff
@@ -1131,18 +1143,53 @@ def generate_stack_component_deploy_command(
         type=str,
     )
     @click.option(
-        "--cloud",
-        "-c",
-        "cloud",
+        "--provider",
+        "-p",
+        "provider",
+        required=True,
         type=click.Choice(["aws", "gcp", "k3d"]),
-        help="The cloud provider to use to deploy the stack component.",
+        help="The cloud (or local provider) to use to deploy the stack component.",
+    )
+    @click.option(
+        "--region",
+        "-r",
+        "region",
+        required=True,
```

Comment:
Does region make any sense for `k3d`?

Reply:
Why do we make region a first-class flag? Can it be passed under the collective flag we use for other attributes? (FRIW, `-x`)

Reply:
This is unfortunately now locked in from the `mlstacks` side where the stack spec requires you to
specify the region when defining a stack (and hence for stack component deploy here).

Reply:
For k3d the region doesn't get used, so docs will suggest using `local` for the
region though you could basically write anything.